nmoscommon/auth/claims_validator.py

```diff
@@ -32,19 +35,51 @@ def validate_aud(self):  # placeholder
 
     def validate_nmos(self):
         claim_name = "x-nmos-api"
-        valid_claim_option = self.options.get(claim_name)
-        valid_claim_value = valid_claim_option.get('value')
+        actual_claim_value = self.get(claim_name)  # actual claim value in JWT
+        valid_claim_option = self.options.get(claim_name)  # from `claim_options.py`
 
-        actual_claim_value = self.get(claim_name)
-        if not actual_claim_value:
+        if not actual_claim_value:  # Missing x-nmos-api claim
             raise MissingClaimError(claim_name)
-        if not valid_claim_value:
+        if not valid_claim_option:  # No options given for validation
             return
 
-        # TODO - add validation for versions and read/write
-        for attr in valid_claim_value:
-            if actual_claim_value.get(attr) != valid_claim_value.get(attr):
-                raise InvalidClaimError(claim_name)
+        def _validate_permissions_object(valid_api_value):
+            access_permission_object = actual_claim_value.get(valid_api_value)
+            if not access_permission_object:
+                raise InvalidClaimError("{}. No entry in claim for '{}'.".format(claim_name, valid_api_value))
+
+            if request.method in ["GET", "OPTIONS", "HEAD"]:
+                access_right = "read"
+            else:
```

Comment:
Might be worth being explicit here about the write methods? If we get an unexpected method we don't want to give any implicit permission.